### PR TITLE
porting new Qdb for ARM and THUMB

### PR DESCRIPTION
**Pull request by ucgJhe**
as title.

### CODE REVIEW

File: qiling/debugger/qdb/utils.py [hunk 1]
```diff
@@ -97,6 +347,9 @@ def _read_reg(regs, _reg):
     ret_addr = cur_addr + MIPS_INST_SIZE
 
     if line.mnemonic.startswith('j') or line.mnemonic.startswith('b'):
+        if line.mnemonic == "break": # indicates program extied
+            return PROGRAM_EXITED
```
Comment (xwings) [inline qiling/debugger/qdb/utils.py:351]: PROGRAM_EXITED maybe IS_EXIT ?

File: qiling/debugger/qdb/utils.py [hunk 2]
```diff
@@ -1,27 +1,15 @@
 #!/usr/bin/env python3
-from qiling.const import *
+
 from functools import partial
 
+from qiling.const import *
 
+PROGRAM_EXITED = "DONE"
```
Comment (xwings) [inline qiling/debugger/qdb/utils.py:7]: DONE ? why not true or false ?

File: qiling/debugger/qdb/core.py [hunk 3]
```diff
@@ -203,6 +211,21 @@ def do_examine(self, args):
...
+        context_reg(self._ql, self._saved_states)
+        context_asm(self._ql, self._ql.reg.arch_pc, 4)
+
+    def do_show(self, *args):
+        """
+        show some runtime informations
+        """
+        self._ql.mem.show_mapinfo()
+        print("Qdb:", [(hex(idx), val) for idx, val in self.breakpoints.items()])
+        print("internal:", [(hex(idx), val) for idx, val in self._ql._addr_hook.items()])
```
Comment (xwings) [inline qiling/debugger/qdb/core.py:227]: i am not too sure. do we need print or we can use nprint. just in case

Comment (ucgJhe) [reply qiling/debugger/qdb/core.py:227]: this information is for debugging qdb and qiling itself mostly , and use native print for not messing up log file

File: qiling/debugger/qdb/utils.py (hunk 2 above)
Comment (ucgJhe) [reply qiling/debugger/qdb/utils.py:7]: this value will be returned as a address, so true or false will be converted to 0x1 or 0x0 by python automatically, will mess up the breakpoint thing , but sure i can change it

**Comment by xwings** [comment]
sweet! Thanks

**Pull request merged**